From 6968adf1c01ce7830a888910fd6371dc549bd8b0 Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Wed, 1 Aug 2018 18:45:55 +0000
Subject: [PATCH 0920/1076] MANTIS 0051132 extension: Conveyor-Comp: additional
 Baggage Error; tub weight ConveyorDev.idl V1.16

added errorTypes
  ERR_IRREGULAR_BAG,
  ERR_BAG_WITH_STRAPS,
  ERR_UPRIGHT_BAG,
  ERR_SLOPED_BAG,
  ERR_VOLUME_NOT_DETERMINABLE,
  ERR_OVERFLOW_TUB

added attribute bagInfo.tubWeight


git-svn-id: svn://localhost/SelfServiceCommon/trunk@1367 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/idl/Massai/ConveyorDev.idl              | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl b/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
index 949dc257..b973d167 100644
--- a/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
+++ b/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
@@ -39,7 +39,7 @@
 **/
 module conveyordev
 {
-   const string interface_version = "1.15";
+   const string interface_version = "1.16";
 /*
 using DeviceComponents::indicationListener and DeviceComponents::Generic from mdci.idl
   RC indicationListener::indication(in any ind);
@@ -187,7 +187,14 @@ using DeviceComponents::indicationListener and DeviceComponents::Generic from md
       ERR_PIECE_MISTRACKED,         /**< Baggage tracking failure. */
       ERR_MANUAL_MODE,              /**< Online, manual mode */
       ERR_SWITCHED_OFF,             /**< Online, switched off */
-      ERR_HARDWARE_DEGRADED         /**< Hardware partial broken */
+      ERR_HARDWARE_DEGRADED,        /**< Hardware partial broken */
+      ERR_IRREGULAR_BAG,            /**< Detected bag should be handled as bulky ERR like Skis, Strollers, Prams,
+                                         Wheelchairs */
+      ERR_BAG_WITH_STRAPS,          /**< Detected bag has lose straps which could block the BHS */
+      ERR_UPRIGHT_BAG,              /**< Detected bag stands upright */
+      ERR_SLOPED_BAG,               /**< Bag leaning/levering on side walls probnably reducing weight */
+      ERR_VOLUME_NOT_DETERMINABLE,  /**< The SBD system cannot perform a volumetric analysis of the bag */
+      ERR_OVERFLOW_TUB              /**< Detected bag overflows the sides of the tub */
    };
 
 
@@ -366,6 +373,9 @@ using DeviceComponents::indicationListener and DeviceComponents::Generic from md
       **/
       long              baggageWeight;
 
+      /** The weight of the tub. The weight is given in units of 1 gramm. **/
+      long              tubWeight;
+
       /**
        *
        * The barcode data read from the currently accepted piece.
-- 
2.41.0.windows.1

